### DISCUSSION TITLE
Declarative configuration

### DISCUSSION THREAD

**Discussion by oneingan** [chrisbenincasa/tunarr#450] [Ideas]
For sharing configurations and for unattended deployments would be huge to have a declarative config mechanism aside the UI. This open the door to massive deployments using Infra-as-code tools like NixOS or Terraform (openTOFU).

**Comment by chrisbenincasa** [comment]
Thanks for the idea! Can you be a bit more specific about what you mean? You mention infra-as-code...are you talking about configuring the server/backend, or do you mean encoding the actual configuration of Tunarr entities (channels, programs, etc) in this manner?

**Reply by oneingan** [reply]
I mean encoding the actual configuration of Tunarr entities (channels, programs, etc).

**Reply by chrisbenincasa** [reply]
Gotcha. That's an interesting idea. Most likely the way we'd tackle it is with a JSON schema, then we'd just have adapters for IaC, etc. I've been exploring a version of this with declarative scheduling for Dynamic Channels (#15). You can see some of preliminary work for that on the branch: https://github.com/chrisbenincasa/tunarr/commits/dynamic-channels/

I think doing this for all entities would be a hefty task, but agree that it would potentially be very powerful. I really like the idea of being able to import configurations from others.
